File: src/main/java/info/dddeurope/lab/datastores/EventQueue.java
package info.dddeurope.lab.datastores;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import info.dddeurope.lab.app.common.Event;

public class EventQueue {

    private Map<String, List<Event>> store = new HashMap<>();

    public void addEvent(String id, Event event) {
        List<Event> events = store.get(id);
        if (events == null) {
            events = new ArrayList<Event>();
            events.add(event);
            store.put(id, events);
        } else {
            events.add(event);
        }
    }

    public List<Event> getEvents(String id) {
        return store.get(id);
    }

}

File: DotNet/App/DataStores/EventQueue.cs
using System.Collections.Generic;
using App.Common;

namespace App.DataStores
{
    public class EventQueue
    {
        private readonly Dictionary<string, List<IEvent>> _store = new();

        public void AddEvent(string id, IEvent evt) {
            if (!_store.TryGetValue(id, out var events)) {
                events = new List<IEvent> {evt};
                _store[id] = events;
            } else {
                events.Add(evt);
            }
        }

        public List<IEvent> GetEvents(string id)
        {
            return _store.TryGetValue(id, out var events) ? events : null;
        }
    }
}